twitter/src/main/java/com/hazelcast/jet/contrib/twitter/TwitterSources.java

```diff
@@ -0,0 +1,201 @@
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+
+import com.twitter.hbc.ClientBuilder;
+import com.twitter.hbc.core.endpoint.StreamingEndpoint;
+import com.twitter.hbc.httpclient.BasicClient;
+import com.twitter.hbc.core.processor.StringDelimitedProcessor;
+import com.twitter.hbc.httpclient.auth.Authentication;
+import com.twitter.hbc.httpclient.auth.OAuth1;
+
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.Properties;
+
+import javax.annotation.Nonnull;
+
+
+/**
+ * Contains methods for creating Twitter stream sources.
+ */
+public final class TwitterSources {
+
+    private TwitterSources() {
+    }
+
+    /**
+     * Creates a {@link StreamSource} which reads tweets from Twitter's Streaming API
+     * for data ingestion to Jet pipelines.
+     * Example usage:
+     * <pre>{@code
+     * Properties credentials = loadTwitterCredentials();
+     * List<String> terms = new ArrayList<String>(Arrays.asList("BTC", "ETH"));
+     * StreamSource<String> timestampedStreamSource =
+     *              TwitterSources.timestampedStream(
+     *                      "twitter-timestampedstream-source",
+     *                      () -> new StatusesFilterEndpoint().trackTerms(terms),
+     *                      credentials,
+     *                      Constants.STREAM_HOST
+     *              );
+     * Pipeline p = Pipeline.create();
+     * StreamSourceStage<String> srcStage = p.readFrom(timestampedStreamSource);
+     * }</pre>
+     *
+     * @param name             a descriptive name of this source.
+     * @param endpointSupplier Supplier that supplies a Twitter StreamingEndpoint to connect to source.
+     * @param credentials      a Twitter OAuth1 credentials that consists "consumerKey",
+     *                         "consumerSecret", "token", "tokenSecret" keys.
+     * @param host             a Twitter host URL to connect.
+     *                         These hosts are defined in {@link com.twitter.hbc.core.Constants}.
+     * @return a timestamped stream source to use in {@link com.hazelcast.jet.pipeline.Pipeline#readFrom}
+     */
+    @Nonnull
+    public static StreamSource<String> timestampedStream(@Nonnull String name,
+                                                         @Nonnull SupplierEx<? extends StreamingEndpoint> endpointSupplier,
+                                                         @Nonnull Properties credentials,
+                                                         @Nonnull String host) {
+        return SourceBuilder.timestampedStream(name,
+                ctx -> new TwitterSourceContext(endpointSupplier, credentials, host))
+                .fillBufferFn(TwitterSourceContext::fillTimestampedBuffer)
+                .destroyFn(TwitterSourceContext::close)
+                .build();
+    }
+
+
+    /**
```

Comment:
instead of duplicating the javadoc maybe we can link the above method and mention that the only difference is that while the above method produces a timestamped source this one produces a source without timestamps